libres/lib/analysis/analysis_module.cpp

```diff
@@ -42,7 +42,7 @@ auto logger = ert::get_logger("analysis");
 
 struct analysis_module_struct {
     UTIL_TYPE_ID_DECLARATION;
-    ies::data::data_type *module_data;
+    std::unique_ptr<ies::data::Data> module_data;
```

Comment:
Why `unique_ptr` and not just?
```suggestion
    ies::data::Data module_data;
```

Reply:
Good question - I'll update

Reply:
No. `unique_ptr` is not a trivially-constructible object so it cannot be used in conjunction with `malloc` (undefined behaviour). Please change the alloc function to use `new`.